### ISSUE TITLE
VAEC Migration Planning

### ISSUE THREAD

**Issue by jwoodman5** [department-of-veterans-affairs/va.gov-team#98621]
## Status
| Date       | Status         | Launch Date   | Notes                             |
|------------|----------------|---------------|-----------------------------------|
| 2025-5-1 | On Hold | TBD | This work is currently on hold 
| 2025-4-18 | In Progress | On Track (Launch Date TBD) | Additional team members have started the ePAS request process, and the Initial high-level migration plan is underway.
| 2025-4-3 | In Progress | On Track (Launch Date TBD) | 2 initial admin accounts have been set up, we'll begin to take the rest of the team through this process next sprint. Initial high level migration plan is underway.
| 2025-3-21 | In Progress | On Track (Launch Date TBD) | Dev environment was successfully provisioned in VAEC. We'll be requesting admin credentials through NMEA and ePAS, and will begin exploratory work and planning this coming sprint.
| 2025-3-7 | In Progress | On Track (Launch Date TBD) | Conducted 2 Intake meetings for VAEC and ECSO. Provisioning seems to be underway, though 2 additional steps will be needed from us. The first through EPAS, once the dev environment is provisioned. And the other an architectural intake form from ECSO, that we'll prioritize for the upcoming sprint.
| 2025-2-24 | In Progress | On Track (Launch Date TBD) | Initial planning is underway; we conducted a cross-team alignment/planning session to understand each team's questions and concerns with this migration. Environment provisioning and access remain the critical blockers, but communication with Robert Ballon on 2/21 suggested this can move forward. |
| 2025-2-10 | In Progress | On Track (Launch Date TBD) | We reviewed the Turbot results this past sprint, and are getting a report from Deepan. The kickoff meeting was rescheduled for 2/18 due to key stakeholders being unavailable. Platform will help to prepare for the kickoff by gathering known info on agenda items. |

## Product Outline
TBD

## High Level User Story/ies
The [VA.gov Home | Veterans Affairs](http://va.gov/)  Platform must migrate critical workloads to VA Enterprise Cloud (VAEC) to ensure enhanced security, modern infrastructure compatibility, and improved performance. This migration involves moving staging and production workloads from their current hosting environment into VAEC’s AWS infrastructure with minimal downtime for veterans and internal teams.

## Hypothesis or Bet
We believe that with proper planning, and all necessary access for technical team members on engineering and security teams, we will be able to migrate va.gov platform workloads from it’s current AWS environment to VAEC’s AWS environment in a way that will:

Maintain current operations and minimize downtime during the transition.

Improve security posture and enable better adherence to compliance requirements

Identify opportunities for streamlining the current tech stack and improving performance during the migration if the schedule allows it.

How might we...
…migrate va.gov workloads into the VAEC environment in a way that ensures continuity of operations, enhanced security, and improved performance benchmarks while maintaining a seamless experience for veterans?

## OKR
- **OCTO 2025 OKR 2: OCTO's platforms are the fastest, most efficient, and most secure way to deliver products at VA.**
  - O2.1: "100% of authentications to our systems and tools (both Veteran-facing and internal) occur using a secure credential."

- **OCTO 2025 O4: OCTO positively influences VA's ability to deliver software products and services faster, safer, and with higher quality.**
  - O4.2: "OCTO has helped resolve/support at least 10 significant engineering issues or products outside our portfolio."

### 1. Technical Onboarding & Access ### 
- [x] Vendor onboarding questions have been answered and documented (no outstanding clarifications).
- [ ] Environments have been provisioned by VAEC
  - [x] dev environment provisioned
  - [ ] staging environment provisioned
  - [ ] sandbox environment provisioned
  - [ ] production environment provisioned
- [ ] All required access (accounts, credentials) is available, confirmed and functioning.
  - [x] dev environment access 
  - [ ] staging environment access
  - [ ] sandbox environment access
  - [ ] production environment access
- [x] A designated Government technical point of contact (POC) is identified, available to answer technical questions, and supports any necessary escalations.

### 2. Migration & Planning ###
- [ ] A phased migration plan exists (covering dev, staging, sandbox, production), approved by stakeholders, and status is visible in GitHub).
- [ ] The plan includes strategies for minimal downtime and continuity of operations (verified by relevant technical teams).
- [ ] Appropriate monitoring and alerting is established on each workload during the migration as appropriate.

### 3. Security & ATO Readiness ###
- [ ] Required security controls are understood, configured, tested, and meet or exceed VA security standards.
- [ ] Compliance documentation (System Security Plan, Incident Response Plan, Disaster Recovery Plan, etc.) is completed and stored, and a Security/Compliance review is completed.
- [ ] Security scans have been conducted on migrated workloads; all high- and medium-severity vulnerabilities are remediated or documented with an action plan.
- [ ] Stakeholder Review & Sign-Off: Security and compliance teams confirm the system is ATO-ready (though final ATO issuance may continue in parallel).

### 4. Infrastructure Compatibility & CI/CD ###
- [ ] Workloads (dev, staging, production, sandbox) have been migrated to VAEC with no critical issues (e.g., no major outages or data loss).
- [ ] CI/CD pipelines (including Terraform for IaC) deploy to VAEC without errors, and automated tests pass in the new environment.
- [ ] Performance benchmarks in VAEC meet or exceed current va.gov platform metrics (e.g., response time, uptime).
- [ ] Key components, dependencies, and licenses function as expected post-migration, with no negative impact to Veteran-facing services.

### 5. Support Integration & Incident Management ###
- [ ] VAEC 24/7 Incident Support and SLA are integrated with va.gov’s support process (recommend a separate step for validating this)
- [ ] Platform support documentation and runbooks are updated to reflect hosting in VAEC, ensuring on-call staff are informed and equipped to follow consistent, up-to-date procedures.

### 6. Documentation & Knowledge Transfer ###
- [ ] Access Requests, A clear, documented process for requesting, modifying, and revoking credentials is in place and validated by the appropriate stakeholders.
- [ ] Architecture diagrams, environment details, and findings from the migration (e.g., troubleshooting steps, lessons learned) are documented and stored in Confluence/GitHub.
- [ ] Knowledge transfer sessions or demos have been completed for any teams responsible for ongoing operations or future enhancements.

### 7. Final Stakeholder Acceptance ###
- [ ] All relevant stakeholders (VA leadership, Security/Compliance, Infrastructure teams, vendor, etc.) have reviewed and confirmed the project meets its objectives and is ready for formal ATO submission.

### 8. Sunsetting/Decommissioning (Post-Acceptance) ###
- [ ] All deprecated workloads or services from the old environment have been archived or shut down (with a rollback window if needed).
- [ ] Old credentials/accounts that will not be migrated are revoked and documented.
- [ ] Configuration references (DNS entries, load balancer settings, firewall rules) related to the old environment are cleaned up or updated.
- [ ] Documentation is updated to reflect the new, active environment only.
